frontend/catalyst/utils/toml.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Module for abstracting which toml_load to use.
+"""
+
+import importlib.util
+
+# TODO:
+# Once Python version 3.11 is the oldest supported Python version, we can remove tomli
+# and rely exclusively on tomllib.
+
+# New in version 3.11
+# https://docs.python.org/3/library/tomllib.html
+tomllib = importlib.util.find_spec("tomllib")
+tomlkit = importlib.util.find_spec("tomlkit")
+# We need at least one of these to make sure we can read toml files.
+if tomllib is None and tomlkit is None:  # pragma: nocover
+    msg = "Either tomllib or tomlkit need to be installed."
+    raise ImportError(msg)
```

Comment:
in our `setup.py`, can we make `tomllib` a conditional dependency if the user has python 3.11, otherwise `tomlkit`?

Reply:
I believe that shouldn't be a problem when running setup.py directly (say by checking the Python version running the setup.py), but I wonder if this will carry over to the wheel (my guess is not). But maybe there is another mechanism to achieve this?

Reply:
We only need the last line as tomllib is part of Python's standard library. Thanks for PEP508, looks great!

Reply:
https://github.com/PennyLaneAI/catalyst/pull/369/commits/d9f23c7b83f33bdff5d65df5a34460cf04f12134

Reply:
awesome, thanks Erick!